spec/spec_helper.rb

```diff
@@ -14,27 +14,55 @@
 Dir['./spec/support/**/*.rb'].each { |f| require f }
 
 ##
-# Load Rails dummy application based on gemfile name substituted by Appraisal
+# Detect Rails/Sinatra dummy application based on gemfile name substituted by Appraisal
 #
 if ENV['APPRAISAL_INITIALIZED'] || ENV['TRAVIS']
   app_name = Pathname.new(ENV['BUNDLE_GEMFILE']).basename.sub('.gemfile', '')
 else
   app_name = 'rails_5'
 end
 
-require File.expand_path("../../spec/app/#{app_name}/config/environment", __FILE__)
-
-APP_RAKEFILE = File.expand_path("../../spec/app/#{app_name}/Rakefile", __FILE__)
+app_framework = %w{rails sinatra}.find { |f| app_name.to_s.include?(f) }
 
 ##
-# Load Rspec
+# Load dummy application and Rspec
 #
-require 'rspec/rails'
+case app_framework
+when 'rails'
+  # Load Rails
+  require File.expand_path("../app/#{app_name}/config/environment", __FILE__)
 
-# Configure
-RSpec.configure do |config|
-  config.fixture_path = File.join(File.dirname(__FILE__), '/fixtures')
+  APP_RAKEFILE = File.expand_path("../app/#{app_name}/Rakefile", __FILE__)
+
+  # Load Rspec
+  require 'rspec/rails'
+
+  # Configure
+  RSpec.configure do |config|
+    config.fixture_path = File.join(File.dirname(__FILE__), '/fixtures')
+  end
+
+when 'sinatra'
+  # Load Sinatra
+  require File.expand_path("../app/#{app_name}/app", __FILE__)
+
+  # Load Rspec
+  require 'rspec'
+
+  # Configure
+  RSpec.configure do |config|
+    config.filter_run_excluding :rails
 
+    def fixture_path
```

Comment:
For preventing to fail current specs, I defined `fixture_path` manually in sinatra test. `fixture_path` is provided by rspec-rails.